cli/src/pcluster/utils.py

```diff
@@ -333,3 +333,33 @@ def get_attr(obj, attributes, default=None):
         if obj is None:
             return default
     return obj
+
+
+def get_placement_group_for_compute_resource(queue, managed_placement_groups, compute_resource):
+    # prefer compute level managed group over named queue level group
+    placement_group = None
+    if compute_resource.networking and compute_resource.networking.placement_group:
+        if is_fully_defined_placement_group(compute_resource.networking.placement_group):
+            placement_group = (
+                compute_resource.networking.placement_group.name or compute_resource.networking.placement_group.id
+            )
+        elif compute_resource.networking.placement_group.enabled:
+            placement_group = managed_placement_groups[f"{queue.name}-{compute_resource.name}"].ref
+    elif queue.networking.placement_group:
+        if is_fully_defined_placement_group(queue.networking.placement_group):
+            placement_group = queue.networking.placement_group.name or queue.networking.placement_group.id
+        elif queue.networking.placement_group.enabled:
+            placement_group = managed_placement_groups[queue.name].ref
+    return placement_group
+
+
+def is_fully_defined_placement_group(placement_group) -> bool:
+    return (placement_group.id or placement_group.name) and (
+        placement_group.enabled or placement_group.is_implied("enabled")
+    )
+
+
+def is_managed_placement_group(placement_group) -> bool:
+    return (placement_group.id or placement_group.name) and (
+        placement_group.enabled or placement_group.is_implied("enabled")
+    )
```

Comment:
Can we move the two functions to two methods of PlacementGroup class in cluster_config.py?

Reply:
Yup, that makes sense to move them.  However I realized that the managed group function also needs to check the queue level config... so may just get rid of that one